metrics/auto_test.go

```diff
@@ -0,0 +1,45 @@
+package metrics
+
+import (
+	"testing"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+func TestPromAdjust(t *testing.T) {
+	testCases := []struct {
+		input, output string
+	}{
+		{"RA.Foo.Bar", "Foo_Bar"},
+		{"", ""},
+		{"RA-FOO-BAR", "RA_FOO_BAR"},
+		{"RA.FOO-BAR", "FOO_BAR"},
+		{"RA.FOO-BAR", "FOO_BAR"},
+		{"RA", "RA"},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.input, func(t *testing.T) {
+			if promAdjust(tc.input) != tc.output {
+				t.Errorf("expected %q, got %q", tc.input, tc.output)
+			}
+		})
+	}
+}
+
+func TestAutoProm(t *testing.T) {
```

Comment:
Should this also test that trying to create the same autoProm for "foo.bar" will memoize properly?